thermal_base/thermal_base.py

```diff
@@ -0,0 +1,1183 @@
+
+    @staticmethod
+    def coordinates_in_poly(polygon_points, frame_shape=(512, 640)):
+        """Returns indices of points within a polygon specified by polygon_points.
+
+        Args:
+            polygon_points (list): list of ordered coordinates.
+        Returns:
+            tuple: Indices of points within a polygon.
+        """
+        mask = np.zeros(frame_shape, dtype=np.uint8)
+        cv.fillPoly(mask, pts=[np.array(polygon_points)], color=255)
+        return np.where(mask > 0)
+
+    @staticmethod
+    def change_emissivity_for_roi(
+        thermal_np,
+        meta,
+        roi_contours,
+        raw_roi_values,
+        indices,
+        new_emissivity=None,
+        ref_temperature=None,
+        atm_temperature=None,
+        np_indices=False,
+    ):
+        """Changes the Emissivity for pixels in a given region (RoI) and returns the new temperature matrix"""
+        if np_indices:
+            meta_x = meta.copy()
+            if new_emissivity not in (None, ""):
+                meta_x["Emissivity"] = float(new_emissivity)
+            if ref_temperature not in (None, ""):
+                meta_x["ReflectedApparentTemperature"] = float(ref_temperature)
+            if atm_temperature not in (None, ""):
+                meta_x["AtmosphericTemperature"] = float(atm_temperature)
+
+            return ThermalImageHelpers.sensor_vals_to_temp(raw_roi_values, **meta_x)
+
+        x, y, w, h = cv.boundingRect(roi_contours[0])
+        temp_array = thermal_np.copy()
+        roi_rectangle = temp_array[y : y + h, x : x + w]
+        roi_rectangle_flat = roi_rectangle.flatten()
+        roi_raw_sensor_np = np.asarray(raw_roi_values)
+
+        if new_emissivity is None:
+            new_emissivity = float(input("Enter new Emissivity: "))
+
+        meta_x = meta.copy()
+        meta_x["Emissivity"] = new_emissivity
+
+        raw_roi_values_thermal_np = ThermalImageHelpers.sensor_vals_to_temp(roi_raw_sensor_np, **meta_x)
+
+        count = 0
+        for i in range(0, len(roi_rectangle_flat)):
+            if i in indices:
+                roi_rectangle_flat[i] = raw_roi_values_thermal_np[count]
+                count += 1
+
+        temp_array[y : y + h, x : x + w] = roi_rectangle_flat.reshape((h, w))
+        return temp_array
+
+    @staticmethod
+    def default_scaling_image(array, cmap=cv.COLORMAP_JET):
+        """Scales temperature vals for the whole image to be clipped to ~min and max around the mid line"""
+        thermal_np = array.copy()
+        mid_thermal_np = thermal_np[10 : thermal_np.shape[0] - 10, (int)(thermal_np.shape[1] / 2)]
+        maximum = np.amax(mid_thermal_np)
+        minimum = np.amin(mid_thermal_np)
+
+        thermal_np[thermal_np > maximum + 10] = maximum + 10
+        thermal_np[thermal_np < minimum - 5] = minimum - 5
+        image = ThermalImageHelpers.get_cmapped_temp_image(thermal_np, colormap=cmap)
+
+        return image, thermal_np
+
+
+class ThermalImageAnnotation:
```

Comment:
Same possibility can be evaluated for all the functions of this class too. You can shift them to an `annotation.py` if you want. I believe the one `@classmethod` is also something that can be changed to a static method or a function without the class grouping.

Reply:
There are class variables that are being modified by static functions. I believe that this change will disrupt the flow a lot. I'm keeping this class in the same file